src/goto-programs/remove_function_pointers.cpp

```diff
@@ -430,6 +430,26 @@ void remove_function_pointerst::remove_function_pointer(
   statistics().source_location=target->source_location;
   statistics() << "replacing function pointer by "
                << functions.size() << " possible targets" << eom;
+
+  // list the names of functions when verbosity is at debug level
+  if(get_message_handler().get_verbosity() >= M_DEBUG)
```

Comment:
I suggest to add a function to the message handler that executes a function when the debug level check succeeds. You could then call, e.g. `debug(print_functions)` where `print_functions` could just be a lambda when it is short.

Reply:
Done: you can now do `conditional_output(debug(), <lambda>)`